src/capture-screenshot.js

```diff
@@ -0,0 +1,145 @@
+const puppeteer = require('puppeteer');
+const {performance} = require('perf_hooks');
+
+const devicePixelRatio = 1.0;
+const maxTimeInSec = 30;
+
+const timeDelta = (start, end) => {
+  return ((end - start) / 1000).toPrecision(3);
+}
+
+const htmlTemplate = (options) => {
+  const {width, height, inputPath, libPort, backgroundColor} = options;
+  return `
+    <html>
+      <head>
+        <meta name="viewport" content="width=device-width, initial-scale="${devicePixelRatio}">
+        <script type="module"
+          src="http://localhost:${libPort}/model-viewer.js">
+        </script>
+        <style>
+          body {
+            margin: 0;
+          }
+          model-viewer {
+            --progress-bar-color: transparent;
+            width: ${width};
+            height: ${height};
+          }
+        </style>
+      </head>
+      <body>
+        <model-viewer
+          style="background-color: ${backgroundColor};"
+          background-color=""
+          id="snapshot-viewer"
+          interaction-prompt="none"
+          src="${inputPath}" />
+      </body>
+    </html>
+  `
+}
+
+const captureScreenshot = async function (options) {
+  const browserT0 = performance.now();
+  const {width, height, outputPath, debug} = options;
+
+  const browser = await puppeteer.launch({
+    args: [
+      '--no-sandbox',
+    ],
+    defaultViewport: {
+      width,
+      height,
+      deviceScaleFactor: devicePixelRatio,
+    },
+    headless: !debug,
+  });
+
+  const page = await browser.newPage();
+
+  page.on('error', (error) => {
+    console.log(`🚨 ${error}`);
+  });
+
+  page.on('console', async (message) => {
+    const args =
+        await Promise.all(message.args().map((arg) => arg.jsonValue()));
+
+    if (args.length) {
+      console.log(`➡️`, ...args);
+    }
+  });
+
+  const browserT1 = performance.now()
+
+  console.log(`🚀 Launched browser (${timeDelta(browserT0, browserT1)}s)`);
+
+  const contentT0 = performance.now();
+
+  const data = htmlTemplate(options);
+  await page.setContent(data, {waitUntil: 'domcontentloaded'});
+
+  const contentT1 = performance.now();
+
+  console.log(`🗺  Loading template to DOMContentLoaded (${timeDelta(contentT0, contentT1)}s)`);
+
+  const renderT0 = performance.now();
+
+  const evaluateError = await page.evaluate(async (maxTimeInSec) => {
+    const modelBecomesReady = new Promise((resolve, reject) => {
+      let timeout;
+      if (maxTimeInSec > 0) {
+        timeout = setTimeout(() => {
+          reject(new Error(
+              `Stop capturing screenshot after ${maxTimeInSec} seconds`));
+        }, maxTimeInSec * 1000);
+      }
+
+      const modelViewer = document.getElementById('snapshot-viewer');
+      modelViewer.addEventListener('poster-dismissed', () => {
```

Comment:
This is the 'hack' model-viewer does to help with the timing of screenshoting a rendered model: https://github.com/google/model-viewer/blob/master/packages/render-fidelity-tools/test/renderers/model-viewer/index.html#L49. Turns out it works pretty well! I ran a suite of models on a slower VM using Docker and managed to get the images rendering correctly where as before they were all skewed to the top left corner or weird sizes.